File: frontend/src/entities/assessment/model/assessment.ts
/** Результат прохождения симулятора-ассессмента (см. backend assessments). */
export interface AssessmentDecision {
    day: number;
    question: string;
    choice: string;
    tone: "good" | "bad" | "neutral";
    cite?: { law: string; ref: string; note: string };
}

export interface AssessmentAchievement {
    id: string;
    title: string;
    tone: "good" | "bad";
    desc: string;
}

export interface AssessmentCreate {
    candidate_name: string;
    grade: string;
    grade_label: string;
    avg: number;
    compliance: number;
    stats: Record<string, number>;
    decisions: AssessmentDecision[];
    achievements: AssessmentAchievement[];
    profile: Record<string, unknown>;
}

export interface Assessment extends AssessmentCreate {
    id: number;
    created_at: string;
}

File: backend/app/schemas/assessment.py
"""AssessmentResult-related Pydantic schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AssessmentCreate(BaseModel):
    candidate_name: str
    grade: str
    grade_label: str
    avg: int
    compliance: int = 0
    stats: dict = {}
    decisions: list = []
    achievements: list = []
    profile: dict = {}


class AssessmentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    candidate_name: str
    grade: str
    grade_label: str
    avg: int
    compliance: int
    stats: dict
    decisions: list
    achievements: list
    profile: dict
    created_at: datetime
